filters/auth/grantcookie.go

```diff
@@ -12,6 +12,33 @@ import (
 	"golang.org/x/oauth2"
 )
 
+type CookieEncoder interface {
```

Comment:
Naming?

Reply:
Do we want cookie encoder or a generic token encoder that works with http.Response and may encode/extract token into/from cookie or e.g. headers?

Reply:
The existing Oauth2 approach here is designed to work with a cookie. Cookie-specific is fine for us unless you think it can be used somewhere else

Reply:
@AtorVdm I've updated CookieEncoder interface, please check if its suitable for the implementation you have in mind.